mlflow/server/handlers.py

```diff
@@ -261,12 +265,67 @@ def initialize_backend_stores(backend_store_uri=None, default_artifact_root=None
     except UnsupportedModelRegistryStoreURIException:
         pass
 
+_STRING_TYPE_DEFAULT_SCHEMA = {"type": "string"}
+_INTEGER_TYPE_DEFAULT_SCHEMA = {"type": "integer"}
+_REQUIRED_PARAM_DEFAULT_SCHEMA = {"required": True}
+
+_DEFAULT_SCHEMAS = {
+    "name": _STRING_TYPE_DEFAULT_SCHEMA,
+    "artifact_location": _STRING_TYPE_DEFAULT_SCHEMA,
+    "tags": {"type": "array", "minItems": 0},
+    "max_results": _INTEGER_TYPE_DEFAULT_SCHEMA,
+    "page_token": _STRING_TYPE_DEFAULT_SCHEMA,
+    "next_page_token": _STRING_TYPE_DEFAULT_SCHEMA,
+    "experiment_id": _STRING_TYPE_DEFAULT_SCHEMA,
+    "experiment_name": _STRING_TYPE_DEFAULT_SCHEMA,
+    "new_name": _STRING_TYPE_DEFAULT_SCHEMA,
+    "timestamp": _INTEGER_TYPE_DEFAULT_SCHEMA,
+    "start_time": _INTEGER_TYPE_DEFAULT_SCHEMA,
+    "end_time": _INTEGER_TYPE_DEFAULT_SCHEMA,
+    "run_id": _STRING_TYPE_DEFAULT_SCHEMA,
+    "key": _STRING_TYPE_DEFAULT_SCHEMA,
+    "step": _INTEGER_TYPE_DEFAULT_SCHEMA,
+    "model_json": _STRING_TYPE_DEFAULT_SCHEMA,
+    "metric_key": _STRING_TYPE_DEFAULT_SCHEMA,
+    "filter": _STRING_TYPE_DEFAULT_SCHEMA,
+    "path": _STRING_TYPE_DEFAULT_SCHEMA,
+    "description": _STRING_TYPE_DEFAULT_SCHEMA,
+    "stages": {"type": "array", "items": _STRING_TYPE_DEFAULT_SCHEMA},
+    "source": _STRING_TYPE_DEFAULT_SCHEMA,
+    "run_link": _STRING_TYPE_DEFAULT_SCHEMA,
+    "version": _STRING_TYPE_DEFAULT_SCHEMA,
+    "order_by": {"type": "array", "items": _STRING_TYPE_DEFAULT_SCHEMA},
+    "archive_existing_versions": {"type": "boolean"}
+}
+
+
+def _build_schema(name, additional_schema = {}):
+
+    schema = copy.deepcopy(_DEFAULT_SCHEMAS.get(name, {}))
+
+    for k, v in additional_schema.items():
+        schema[k] = v
+
+    return schema
+
+def _validate_param_against_schema(schema, param):
+    ## If the schema is empty, do nothing
+    if not schema:
+        return True
+
+    try:
+        Draft202012Validator(schema).validate(param)
+    except ValidationError as e:
+        raise MlflowException(
+            message=e.message,
+            error_code=INVALID_PARAMETER_VALUE
+        )
```

Comment:
Design nits: This will return the exception that the validator throws as the message, which I think is okay and probably easier than returning some custom message.

Separately, I don't love throwing an error here as a side effect, but sometimes returning `True`. I can't think of a better pattern though. The only other thing that comes to mind would be to either return the error message or a placeholder like "ok", and then check if the return of this call is `"ok"` and pass, otherwise raise the exception. Open to ideas on this one